# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1424625252 -3600
#      Sun Feb 22 18:14:12 2015 +0100
# Node ID 448ca112a5afb5d690cd51f8f3b62d017207b12a
# Parent  ced2313b8d56b8d7ae91e4467698e5a487fed95a
prevent conflicts between run-file and debug-continue action in gui

* file-editor.cc (handle_enter_debug_mode): remove shortcut for run action;
  (handle_exit_debug_mode): restore shortcut for run action

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -136,22 +136,24 @@ file_editor::toolbar (void)
 {
   return _tool_bar;
 }
 
 void
 file_editor::handle_enter_debug_mode (void)
 {
   _run_action->setEnabled (false);
+  _run_action->setShortcut (QKeySequence ());
 }
 
 void
 file_editor::handle_exit_debug_mode (void)
 {
   _run_action->setEnabled (true);
+  shortcut_manager::set_shortcut (_run_action, "editor_run:run_file");
 }
 
 void
 file_editor::request_new_file (const QString& commands)
 {
   // Custom editor? If yes, we can only call the editor without passing
   // some initial contents and even without being sure a new file is opened
   if (call_custom_editor ())
